### PR TITLE
OEL-1248: Add events listing facets

### PR DESCRIPTION
**Pull request by GilNovacomm**
### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesEventTest.php [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+use Behat\Mink\Element\NodeElement;
+use Drupal\node\Entity\Node;
+use Drupal\node\NodeInterface;
+use Drupal\Tests\oe_showcase\ExistingSite\ShowcaseExistingSiteTestBase;
+use Drupal\Tests\search_api\Functional\ExampleContentTrait;
+
+/**
+ * Tests Event list pages.
+ */
+class ListPagesEventTest extends ShowcaseExistingSiteTestBase {
```
Comment (drishu) [inline modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesEventTest.php]: for the testing, since we made changes to the scope, we should rather change ListPagesTest (where listings are already tested) and just 

1. add one event node,
2. when asserting the news items, now, also assert that ony news items are present
3. add another listing page but for events
4. assert the event items, and that ony those are present

**Review by drishu — CHANGES_REQUESTED** [review]

File: modules/oe_showcase_list_pages/oe_showcase_list_pages.info.yml [hunk 2]
```diff
@@ -6,11 +6,17 @@ package: OpenEuropa Showcase
 dependencies:
   - oe_whitelabel:oe_whitelabel_list_pages
   - oe_content:oe_content
+  - oe_starter_content:oe_starter_content_news
+  - oe_starter_content:oe_starter_content_event
```
Comment (drishu) [inline modules/oe_showcase_list_pages/oe_showcase_list_pages.info.yml]: sort these lines

**Review by drishu — DISMISSED** [review]

File: modules/oe_showcase_list_pages/oe_showcase_list_pages.info.yml (hunk 2 above)
Comment (donquixote) [reply modules/oe_showcase_list_pages/oe_showcase_list_pages.info.yml]: e before n

File: modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php [hunk 3]
```diff
@@ -81,42 +111,100 @@ public function testCreateListPages() {
...
+   *   Expected List page content type title.
    * @param int $expected_count
    *   Expected number of results to be reported in the title.
    */
-  protected function assertSearchResultsTitle(int $expected_count): void {
+  protected function assertSearchResultsTitle(string $listing_page_content, int $expected_count): void {
     $title = $this->assertSession()->elementExists('css', '.col-xxl-8 h4.mb-0');
     $this->assertSame(
-      sprintf('News list page (%s)', $expected_count),
+      sprintf('%s list page (%s)', $listing_page_content, $expected_count),
```
Comment (donquixote) [inline modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php]:
```suggestion
      sprintf('%s (%s)', $list_page_title, $expected_count),
```

File: modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php [hunk 4]
```diff
@@ -81,42 +111,100 @@ public function testCreateListPages() {
...
   /**
    * Asserts the title above the search results.
    *
+   * @param string $listing_page_content
+   *   Expected List page content type title.
    * @param int $expected_count
    *   Expected number of results to be reported in the title.
    */
-  protected function assertSearchResultsTitle(int $expected_count): void {
+  protected function assertSearchResultsTitle(string $listing_page_content, int $expected_count): void {
```
Comment (donquixote) [inline modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php]:
```suggestion
  protected function assertSearchResultsTitle(string $list_page_title, int $expected_count): void {
```

File: modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php [hunk 5]
```diff
@@ -81,42 +111,100 @@ public function testCreateListPages() {
...
+    $title_input->setValue('News');
+    $search_button->click();
+    $this->assertSearchResultsTitle('Event', 0);
   }
 
   /**
    * Asserts the title above the search results.
    *
+   * @param string $listing_page_content
+   *   Expected List page content type title.
```
Comment (donquixote) [inline modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php]: First, the parameter name `$listing_page_content` is weird.
Second, why not pass the complete node title of the list page?

```suggestion
   * @param string $list_page_title
   *   Title of the list page node.
```

File: modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php [hunk 6]
```diff
@@ -81,42 +111,100 @@ public function testCreateListPages() {
...
     $this->assertNotNull($publication_date_input);
 
-    // Filter results by date.
+    // Filter the News results by date.
     $publication_date_input->setValue('gt');
     $date_input = $filter_form->findField('Date');
     $date_input->setValue('2022-04-04');
     $search_button->click();
-    $this->assertSearchResultsTitle(8);
+    $this->assertSearchResultsTitle('News', 6);
```
Comment (donquixote) [inline modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php]: As suggested below, let's pass the entire node title to the method.
```suggestion
    $this->assertSearchResultsTitle('News list page', 6);
```

File: modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php [hunk 7]
```diff
@@ -81,42 +111,100 @@ public function testCreateListPages() {
...
-      'News number 10',
-      'News number 11',
     ]);
 
-    // Filter results by title.
+    // Filter the News results by title.
     $title_input->setValue('News number 8');
     $search_button->click();
-    $this->assertSearchResultsTitle(1);
+    $this->assertSearchResultsTitle('News', 1);
```
Comment (donquixote) [inline modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php]: As suggested let's pass the entire title to the method.
```suggestion
    $this->assertSearchResultsTitle('News list page', 1);
```

File: modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php [hunk 8]
```diff
@@ -81,42 +111,100 @@ public function testCreateListPages() {
...
+    $search_button->click();
+    $this->assertSearchResultsTitle('Event', 1);
+    $this->assertSearchResults([
+      'Event example',
+    ]);
+
+    // Assert only Event nodes are part of the result.
+    $title_input->setValue('News');
+    $search_button->click();
+    $this->assertSearchResultsTitle('Event', 0);
```
Comment (donquixote) [inline modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php]:
```suggestion
    $this->assertSearchResultsTitle('Event list page', 0);
```

File: modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php [hunk 9]
```diff
@@ -81,42 +111,100 @@ public function testCreateListPages() {
...
+      'Event example',
+    ]);
+
+    // Filter Event results by title.
+    $title_input->setValue('Event');
+    $search_button->click();
+    $this->assertSearchResultsTitle('Event', 1);
+    $this->assertSearchResults([
+      'Event example',
+    ]);
```
Comment (donquixote) [inline modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php:220]: We are not filtering anything here if there is only one event and it has "Event" in the title..

File: modules/oe_showcase_list_pages/config/install/facets.facet.oelp_oe_sc_event__content_type.yml [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    weights:
+      sort: 40
+    settings:
+      sort: ASC
+  url_processor_handler:
+    processor_id: url_processor_handler
+    weights:
+      pre_query: 50
+      build: 15
+    settings: {  }
```
Comment (donquixote) [inline modules/oe_showcase_list_pages/config/install/facets.facet.oelp_oe_sc_event__content_type.yml]: I tried and we don't need this facet.
We only need the content type field in the search index, that's enough.

File: modules/oe_showcase_list_pages/config/install/facets.facet.oelp_oe_sc_news__content_type.yml [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    weights:
+      sort: 40
+    settings:
+      sort: ASC
+  url_processor_handler:
+    processor_id: url_processor_handler
+    weights:
+      pre_query: 50
+      build: 15
+    settings: {  }
```
Comment (donquixote) [inline modules/oe_showcase_list_pages/config/install/facets.facet.oelp_oe_sc_news__content_type.yml]: I tried and we don't need this facet.
We only need the content type field in the search index, that's enough.

File: modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php [hunk 12]
```diff
@@ -40,12 +40,12 @@ public function testCreateListPages() {
     $assert_session = $this->assertSession();
     $page = $this->getSession()->getPage();
 
-    // Create some test nodes.
-    for ($i = 0; $i < 12; $i++) {
+    // Create some News test nodes.
+    for ($i = 0; $i < 10; $i++) {
```
Comment (donquixote) [inline modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php]: Why reduce it to 10 nodes?
With 12 nodes we can also test pagination.

File: modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php [hunk 13]
```diff
@@ -81,42 +111,100 @@ public function testCreateListPages() {
...
+    $page->checkField('emr_plugins_oe_list_page[wrapper][exposed_filters][oelp_oe_sc_event__title]');
+    $page->checkField('emr_plugins_oe_list_page[wrapper][exposed_filters][oelp_oe_sc_event__oe_sc_event_dates]');
+    $page->pressButton('Save');
+
+    $this->drupalGet('node/' . $node->id());
+
+    // Assert that the unique Event item is displayed.
+    $this->assertSearchResults([
+      'Event example',
+    ]);
```
Comment (donquixote) [inline modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php:186]: If we have only a single event, we need to test 2x for date: One time with a date where the result will be empty, another time with a date where the existing event shows in the result.

This said, why do we only have one event? With multiple events we could test it all in one go.

**Review by donquixote — CHANGES_REQUESTED** [review]

File: modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php (hunk 12 above)
Comment (GilNovacomm) [reply modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php]: I've restored the number of items to test also pagination.

File: modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php (hunk 13 above)
Comment (GilNovacomm) [reply modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php:186]: I've followed the same number of Event nodes as for News to test filtering and pagination.

File: modules/oe_showcase_list_pages/oe_showcase_list_pages.info.yml [hunk 14]
```diff
@@ -6,15 +6,13 @@ package: OpenEuropa Showcase
 dependencies:
   - oe_whitelabel:oe_whitelabel_list_pages
   - oe_content:oe_content
-  - oe_starter_content:oe_starter_content_news
   - oe_starter_content:oe_starter_content_event
+  - oe_starter_content:oe_starter_content_news
 
 config_devel:
   install:
-    - facets.facet.oelp_oe_sc_event__content_type
```
Comment (donquixote) [inline modules/oe_showcase_list_pages/oe_showcase_list_pages.info.yml]: This would have been part of the other commit but ok :)

**Review by donquixote — APPROVED** [review]

**Review by drishu — APPROVED** [review]
